gql/utils.go

```diff
@@ -104,14 +106,20 @@ func fieldName(field reflect.StructField) string {
 			// for details on this behavior see https://golang.org/pkg/encoding/json/#Marshal
 			return ""
 		}
-		if name != "" {
+		if name != "" && nameIsValidGraphQL(name) {
```

Comment:
Though I was thinking putting it into this if now that I look at it the squashing of `_` hasn't happened yet which means some names which would be valid after that are being skipped. Perhaps best to do the strings.Replace to a variable and then do the isValid check on that.

Reply:
👍